model/giturl.go

```diff
@@ -0,0 +1,50 @@
+package model
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+)
+
+type GitUrl struct {
+	Host   string
+	Org    string
+	Repo   string
+	Branch string
+	Path   string
+}
+
+func (git *GitUrl) GetCloneInfo() (url, branch string) {
+	return fmt.Sprintf("git@%+s:%+s/%+s.git", git.Host, git.Org, git.Repo), git.Branch
+}
+
+func New(gitUrlStr string) (*GitUrl, error) {
+	// This would match something like git@github.com:org/repo.git/path#branch
+	// path and branch are optional. If not given, default values are "" and "master"
+	gitUrlRegex, _ := regexp.Compile("^git@([^/:#]+):([^/:#]+)/+([^/:#]+)\\.git(/[^#]*)?(#.+)?")
```

Comment:
**EDIT:** Never mind, I did not realize that the argument is treated as an interpreted string literal. 

I suppose an alternative is to use raw string literals, \` \`, if we want to use a conventional regex expression.

----
I believe this regex may be invalid according to regexr.

Here's an alternative that may work:
`^git@([^:]+):([^/]+)\/([^#/]+)(?:\.git)?\/?([^#]*?)(#.+)?$`

https://regexr.com/3p9fp